Frameworks/CoreGraphics/CGContextCairo.mm

```diff
@@ -1629,6 +1630,229 @@
     UNLOCK_CAIRO();
 }
 
+void CGContextCairo::_CGContextDrawStrokedPath(cairo_path_t* flatPath, int i, int prev, int total, CGFloat lineWidth) {
+    cairo_path_data_t* data = &flatPath->data[i];
+    int nextIndex = i + flatPath->data[i].header.length;
+    bool moreData = nextIndex < total;
+    // Try to get next and previous types.
+    cairo_path_data_type_t nextType = CAIRO_PATH_CLOSE_PATH;
+    cairo_path_data_type_t prevType = CAIRO_PATH_CLOSE_PATH;
+    if (moreData) {
+        nextType = flatPath->data[nextIndex].header.type;
+    }
+    if (i > 0) {
+        prevType = flatPath->data[prev].header.type;
+    }
+    switch (data->header.type) {
```

Comment:
>switch [](start = 4, length = 6)

This function does too many things, has too many locals and too little comment, so it is hard to exactly understand what is going on.  Could you please:

1. Refactor the code here and break this function down into smaller pieces based on the path element type.
2. Add comments as needed.